library/src/scala/runtime/LazyVals.scala

```diff
@@ -100,18 +144,35 @@ object LazyVals {
   }
 
   def getOffset(clz: Class[_], name: String): Long = {
-    val r = unsafe.objectFieldOffset(clz.getDeclaredField(name))
     if (debug)
-      println(s"getOffset($clz, $name) = $r")
+      clz.getDeclaredFields.nn.foreach(println(_))
+    val field = clz.getDeclaredField(name)
+    if java.lang.reflect.Modifier.isStatic(field.nn.getModifiers()) then
+      unsafe.staticFieldOffset(field)
```

Comment:
Not sure if this is compatible with code compiled with `3.0`. Could would add a case where this would be needed in the new implementation to `tests/run/lazyVals_c3.0.0.scala` and `tests/run/lazyVals_c3.1.0.scala`?

Reply:
This PR also adds getStaticOffset and the compiler should know which fields are static, so couldn't it emit calls to getStaticOffset when needed instead of checking this at runtime?